docs/style.md

```diff
@@ -269,3 +281,191 @@ When writing event handlers, we prefer using the exact name, `onClick` to a
+
+When authoring the styles for a component, it's important that we use
+[`mixins`](https://sass-lang.com/documentation/at-rules/mixin) to allow
+developers to control when the CSS for a specific component gets emitted. For
+example:
+
+```scss
+// src/components/accordion/_accordion.scss
+
+/// Accordion
+/// @access private
+/// @group accordion
+@mixin accordion {
+  .#{$prefix}--accordion {
+    // ...
+  }
+}
+```
+
+Authoring component styles under a mixin allows the design system to:
+
+- Control when the CSS for accordion gets emitted
+- Allows us to author experimental or future styles in a separate mixin and
+  toggle its inclusion through feature flags
+- Could allow developers consuming the design system to control when styles get
+  emitted
+
+#### Use design tokens where appropriate
+
+We have a number of Sass variables available in the project to be used as design
+tokens when building out a component. Almost always you will want to leverage
+these instead of hard coding values for colors, type, or even spacing. You can
+visit the following SassDoc links to view all of the design tokens relevant to
+this project:
+
+- [Color](../packages/theme/docs/sass.md)
+- [Layout](../packages/layout/docs/sass.md)
+- [Motion](../packages/motion/docs/sass.md)
+- [Type](../packages/type/docs/sass.md)
+
+#### Avoid nesting selectors
+
+Nesting selectors is often a convenient and fast way to author styles in Sass.
+Unfortunately, they also add a performance and maintenance burden for the design
+system. The performance burden is due to the generated nature of selectors which
+can lead to unexpected CSS bundle bloat. The maintenance burden manifests itself
+in a way that makes it harder to find specific selectors while working on the
+codebase. For example, if we are looking for the selector `.component:focus` in
+the following file:
+
+```scss
+// Early on in the file
+.component {
+  // ...
+}
+
+// ...
+
+// Later on in the file
+.component {
+  &:focus {
+    // ...
+  }
+}
+```
+
+It can be difficult to track down the specific `.component:focus` selector
+without having to navigate through the file and relevant matches to see where
+`&:focus` is being defined. While this may be hard to visualize in a short code
+snippet, as file size grows and our Sass is rewritten or updated, this problem
+becomes increasingly obvious.
+
+#### Use only as much specificity as needed
+
+It's important that we write selectors that use as much specificity as needed.
+Ideally, we would only need one selector per component but this is rarely the
+case. As a result, adding specificity should be done sparingly or when including
+it is helpful for building a component. For example, if you would like to
+enforce a specific element or ARIA attribute using this attribute in a selector
+would be appropriate:
+
+```scss
+button[aria-expanded='false'] {
+  // ...
+}
+```
+
+While this example may contain more specificity than needed, it's more important
```

Comment:
I understand the purpose behind this example but I wonder if this `button[aria-expanded='false']` selector is really "more specificity than needed" given the context. 😅 To me it is the opposite: it is seems like the right amount of specificity (no more, no less).

So maybe the sentence could be rephrased --

```suggestion
This specificity is important to ensure
```

Reply:
@jendowns Would it help to contrast it with `.my-component__button` as a direct handle? I think the `button[aria-expanded='false']` one will end up winning out:

![image](https://user-images.githubusercontent.com/3901764/72453194-51742b80-3784-11ea-820f-44c53cb091db.png)

Which is why I said it could be more than needed

Reply:
Yes @joshblack I think that would make the example more clear! 👍